runtime/jvmti/jvmtiExtensionMechanism.c

```diff
@@ -3869,12 +3867,14 @@ jvmtiGetCarrierThread(jvmtiEnv* jvmti_env, ...)
 
 		rc = getVMThread(
 				currentThread, virtual_thread, &targetThread, JVMTI_ERROR_NONE,
-				J9JVMTI_GETVMTHREAD_ERROR_ON_DEAD_THREAD | J9JVMTI_GETVMTHREAD_ERROR_ON_NULL_JTHREAD);
+				J9JVMTI_GETVMTHREAD_ERROR_ON_DEAD_THREAD);
 		if (JVMTI_ERROR_NONE == rc) {
-			if (NULL != targetThread->carrierThreadObject) {
+			j9object_t threadObject = targetThread->threadObject;
+			j9object_t carrierThreadObject = targetThread->carrierThreadObject;
+			if ((NULL != carrierThreadObject) && (threadObject != carrierThreadObject)) {
```

Comment:
Why the second check?

Reply:
`(threadObject != carrierThreadObject)`: The second reflects that a virtual thread is mounted.

Reply:
So if nothing is mounted on the carrier, you won't return the carrier thread object? That seems odd.

Reply:
Ah I suppose we're asuming the input here is a vthread, so I suppose this makes sense.